robotoff/app/schema.py

```diff
@@ -40,3 +40,48 @@
     },
     "required": ["value", "type"],
 }
+
+PREDICT_CATEGORY_SCHEMA: JSONType = {
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "Predict Category",
+    "anyOf": [
+        {
+            "type": "object",
+            "properties": {
+                "barcode": {
+                    "type": "string",
+                    "minLength": 1,
+                },
+                "deepest_only": {
+                    "type": "boolean",
+                },
+                "threshold": {"type": "number"},
+            },
+            "required": ["barcode"],
+        },
+        {
+            "type": "object",
+            "properties": {
+                "product": {
+                    "type": "object",
+                    "properties": {
+                        "product_name": {
+                            "type": "string",
+                            "minLength": 1,
+                        },
+                        "ingredients_tags": {
+                            "type": "array",
+                            "items": {"type": "string"},
+                        },
+                    },
+                    "required": ["product_name"],
+                },
+                "deepest_only": {
+                    "type": "boolean",
+                },
+                "threshold": {"type": "number"},
+            },
+            "required": ["product"],
+        },
+    ],
+}
```

Comment:
It's a bit of a pity we cannot reuse the api.yml.

I'm asking myself if it would be the case to load api.yml at module load time, and reuse sub parts ?

Reply:
Indeed it's a bit of code duplication... Most routes however don't accept JSON as content